llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -10348,6 +10346,14 @@ SDValue RISCVTargetLowering::lowerVECTOR_REVERSE(SDValue Op,
     GatherOpc = RISCVISD::VRGATHEREI16_VV_VL;
   }
 
+  // At LMUL > 1, do the index computation in 16 bits to reduce register
+  // pressure.
+  if (IntVT.getScalarType().bitsGT(MVT::i16) && isUInt<16>(MaxVLMAX - 1) &&
```

Comment:
> lagest VLMAX of 65,536 occurs for LMUL=8 and SEW=8 with VLEN=65,536

According to spec, we don't need `isUInt<16>(MaxVLMAX - 1)` here.

Reply:
Good point. I think we can also remove it from the check added in #66291 then